training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -25,11 +30,35 @@
     @Autowired
     private BookRepository bookRepository;
 
+    @Autowired
+    private OpenLibraryService openLibraryService;
+
     @GetMapping
     public Iterable findAll() {
         return bookRepository.findAll();
     }
 
+
+    public Book getBookOfService(String isbn) {
+        try {
+            Book bookService = openLibraryService.bookInfo(("ISBN:").concat(isbn))
+                .orElseThrow(BookNotFoundException::new);
+            return bookRepository.save(bookService);
+        } catch (ParseException e) {
+            throw new ParseErrorException();
+        } catch (JsonProcessingException e) {
+            throw new JsonProcessingErrorException();
+        }
+
+    }
```

Comment:
The controller must not contain business logic. The business logic must be in the services. These `catch`es and `throw`s should be in the open library service.